wordpress-mobile/WordPress-iOS
# Dashboard Cards: Pages card skeleton

Depends on #20477
Closes #20439 
Part of #20438 

## Description

Adds a skeleton Pages card to the dashboard. The PR also adds some logic and tests related to handling changes to the dashboard endpoint.

## Screenshot

<img src="https://user-images.githubusercontent.com/25306722/229987284-d8b6af46-2dc6-4330-bbf6-e081e71296c4.png" width=35%>

## Testing Instructions

1. Run the Jetpack app
2. Enable the Pages Card Remote Feature Flag from the debug Menu
3. Enable the Personalize Home Tab Feature Flag from the debug Menu
4. Navigate to the dashboard
5. Make sure that the Pages card is added in the correct position
6. Tap on "Personalize your home tab"
7. Make sure the Pages card shows up
8. Make sure that hiding and reshowing the new card works as expected

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added tests to `BlogDashboardServiceTests` and `DashboardCardTests`

PR submission checklist:
